scripts/run_mesh_experiments.py

```diff
@@ -0,0 +1,293 @@
+    vocabulary_size = params["vocabulary_size"]
+    sequence_length = params["sequence_length"]
+    texts, tags = create_dataset(
+        data_path, nb_tags, nb_examples_per_tag)
+
+    nb_train = len(texts) - min(int(0.2*len(texts)), 10_000)
+    train_texts = texts[:nb_train]
+    train_tags = tags[:nb_train]
+    test_texts = texts[nb_train:]
+    test_tags = tags[nb_train:]
+
+    X_train, X_test, Y_train, Y_test = vectorize_data(
+        train_texts, train_tags, test_texts, test_tags,
+        vocabulary_size, sequence_length)
+
+    f1, steps = train(X_train, X_test, Y_train, Y_test, params)
+
+    run_datetime = datetime.now().strftime("%Y%m%d-%H%M%S")
+    with tf.summary.create_file_writer(f"logs/hparam_tuning/run-{run_datetime}".format()).as_default():
+        hp.hparams(params)
+        tf.summary.scalar('f1', f1, step=1)
+        tf.summary.scalar('steps', steps, step=1)
+    return f1, steps
+
+def run_mesh_experiments(data_path, learning_rate, batch_size, nb_tags, nb_examples_per_tag,
+                         dropout, l2, hidden_size, dense_size, attention, architecture,
+                         epochs, vocabulary_size, sequence_length):
+    params = {
+        "learning_rate": learning_rate,
+        "batch_size": batch_size,
+        "nb_tags": nb_tags,
+        "nb_examples_per_tag": nb_examples_per_tag,
+        "dropout": dropout,
+        "l2": l2,
+        "hidden_size": hidden_size,
+        "dense_size": dense_size,
+        "attention": attention,
+        "architecture": architecture,
+        "epochs": epochs,
+        "vocabulary_size": vocabulary_size,
+        "sequence_length": sequence_length
+    }
+    if type(learning_rate) is str and ',' in learning_rate:
+        for learning_rate in learning_rate.split(','):
+            params["learning_rate"] = float(learning_rate)
+            experiment(data_path, params)
+
+if __name__ == '__main__':
+    argparser = ArgumentParser(description=__doc__.strip())
+    argparser.add_argument('--data_path', type=Path)
+    argparser.add_argument('--learning_rate', default=0.01)
+    argparser.add_argument('--batch_size', type=int, default=32)
+    argparser.add_argument('--nb_tags', type=int, default=32)
+    argparser.add_argument('--nb_examples_per_tag', type=int, default=100)
+    argparser.add_argument('--dropout', type=float, default=0.1)
+    argparser.add_argument('--l2', type=float, default=1e-7)
+    argparser.add_argument('--hidden_size', type=int, default=100)
+    argparser.add_argument('--dense_size', type=int, default=10_000)
+    argparser.add_argument('--attention', type=bool, default=True)
+    argparser.add_argument('--architecture', default="cnn")
+    argparser.add_argument('--epochs', type=int, default=500)
+    argparser.add_argument('--vocabulary_size', type=int, default=400_000)
+    argparser.add_argument('--sequence_length', type=int, default=400)
+    argparser.add_argument('--config', type=Path)
+    args = argparser.parse_args()
+
```

Comment:
So you have two ways of submitting a model - via the config file or via setting the parameters yourself. I see, that sounds good.

Reply:
I do that in all my scripts, nothing new :)